scripts/src/java/org/oppia/android/scripts/todo/TodoCollector.kt

```diff
@@ -0,0 +1,95 @@
+package org.oppia.android.scripts.todo
+
+import org.oppia.android.scripts.common.RepositoryFile
+import org.oppia.android.scripts.proto.TodoExemptions
+import java.io.File
+import java.io.FileInputStream
+
+/** Helper class to collect all the TODOs in the repository. */
+class TodoCollector {
+  companion object {
+    private val todoDetectorRegex = Regex(
+      pattern = "(?!\".*)\\bTODO\\b(?!.*\")",
+      option = RegexOption.IGNORE_CASE
+    )
+
+    private val todoExemptionTextProto = "scripts/assets/todo_exemptions"
+
+    // List of all the false todos.
+    private val todoExemptionList = loadTodoExemptionsProto(todoExemptionTextProto)
+      .getTodoExemptionList()
+
+    /**
+     * Collects all the TODOs in the repository.
+     *
+     * @param repoPath the path to the repository
+     * @return a list of all the TODOs
+     */
+    fun collectTodos(repoPath: String): List<Pair<File, Int>> {
+      val searchFiles = RepositoryFile.collectSearchFiles(repoPath)
+      val allTodos = searchFiles.flatMap { file ->
+        file.bufferedReader()
+          .lineSequence()
+          .mapIndexedNotNull { lineIndex, lineContent ->
+            findTodoInFile(file, lineContent, lineIndex, repoPath)
+          }
+      }
+      return allTodos
+    }
+
+    /**
+     * Searches for a TODO at a specific line of the file.
+     *
+     * @param file the file containing the TODOs
+     * @param lineContent the line string
+     * @param lineIndex the index of the line sequence which is to be searched for a TODO
+     * @param repoPath the path to the repository
+     * @return a pair of file and line index for the matched TODO. If not matched, it returns null
+     */
+    private fun findTodoInFile(
+      file: File,
+      lineContent: String,
+      lineIndex: Int,
+      repoPath: String
+    ): Pair<File, Int>? {
+      val relativeFilePath = RepositoryFile.retrieveRelativeFilePath(file, repoPath)
+      if (isTodoNonReal(relativeFilePath, lineIndex + 1)) {
+        return null
+      }
+      return if (todoDetectorRegex.containsMatchIn(lineContent)) Pair(file, lineIndex) else null
+    }
+
+    /**
+     * Checks whether a detected TODO is non-real.
```

Comment:
What does non-real mean? That sounds like a mathematical term, but I don't think that's what you mean here. Can you rephrase & rename to clarify?

Reply:
This is outdated.